### ISSUE TITLE
[Burn-P3+ Bug]: Fuels restricted in all not caught in individual seasons.

### ISSUE THREAD

**Issue by BadgerOnABike** [BurnP3/BurnP3Plus#28]
### Contact Details

brett.moore@nrcan-rncan.gc.ca

### What happened?

When we assess the restricted fuels, the "All" season is its own season rather than applying to all seasons.

https://github.com/BurnP3/BurnP3Plus/blob/3baf6d32e6fad1b44b52c843e58f1f21ed438abb/src/ignitions.R#L222

### What component are you seeing the problem on?

R

### Relevant log output

```
restrictedFuels <- IgnitionRestriction %>%
filter(Season == season | is.na(Season), Cause == cause | is.na(Cause), FireZone == firezone | is.na(FireZone)) %>% 
pull(FuelType)

restrictedFuels
[1] "Matted Grass"

IgnitionRestriction
  Iteration Timestep                FuelType Season Cause FireZone
1        NA       NA                Hydrogen    All    NA       NA
2        NA       NA            Matted Grass Spring    NA       NA
3        NA       NA Scrunched-up newspapers Summer    NA       NA
```

### Approvals Process

- [ ] Testing For Issue
- [ ] Merge

**Comment by shreeramsenthi** [comment]
good catch! I was mostly conceptualizing the "All" season as something for the back end to fill in to Season as a clearer alternative to "NA" in the output, but didn't consider the user filling in "All" as an explicit alternative to NA. This might not be the only place that has this issue.

**Comment by BadgerOnABike** [comment]
Perhaps we have it say all where blanks exist for season to make it explicit.

I agree that this may be a larger change than just the 1 spot I've noted it. I have some other testing to do to see what really needs to be fixed and the expected behaviours but those will wait until after the course.

**Comment by BadgerOnABike** [comment]
So far the issues uncovered are:

** I will edit this as I find more **

- [ ] - All is categorized as its own season
- [ ] - All being a season inhibits ignition grids across all seasons from being enabled

**Comment by shreeramsenthi** [comment]
Sorry this totally fell off my radar! I like this fix, the only unexpected thing is that it introduces the special case of the "All" weather zones and fire zones. I think this is a neat idea, but it doesn't currently exist as a built-in record the way the "All" season is.

Personally, I'm in favour of keeping this in as is since anyone who knows about it can leverage it until we get around to implementing it fully if we want to.
